packages/node_modules/@webex/plugin-meetings/test/unit/spec/meetings/index.js

```diff
@@ -0,0 +1,321 @@
+          const spy = sinon.spy(spark.meetings, 'destroy');
+          spark.meetings.listenInternal();
+          Events.emit('DESTROY_MEETING_DECLINE_1_1', {meeting: {id: 'test'}, response: 'test', type: 'test'});
+          assert.calledOnce(spy);
+          done();
+        });
+      });
+      describe('#handleLocusMercury', () => {
+        beforeEach(() => {
+          spark.meetings.handleLocusEvent = sinon.stub().returns(true);
+        });
+        it('doesnt call handle locus mercury for a locus roap event', () => {
+          spark.meetings.handleLocusMercury({data: {eventType: 'locus.message.roap'}});
+          assert.notCalled(spark.meetings.handleLocusEvent);
+        });
+        it('doesnt call handle locus mercury for an undefined eventType', () => {
+          spark.meetings.handleLocusMercury({data: {}});
+          assert.notCalled(spark.meetings.handleLocusEvent);
+        });
+        it('calls handle locus mercury for all locus events', () => {
+          spark.meetings.handleLocusMercury({data: {eventType: 'test'}});
+          assert.calledOnce(spark.meetings.handleLocusEvent);
+          assert.calledWith(spark.meetings.handleLocusEvent, {eventType: 'test'});
+        });
+      });
+      describe('#handleLocusEvent', () => {
+        describe('there was a meeting', () => {
+          let parse;
+          beforeEach(() => {
+            parse = sinon.stub().returns(true);
+            MeetingCollection.getByKey = sinon.stub().returns({locusInfo: {parse}});
+          });
+          it('should parse the meeting info', () => {
+            spark.meetings.handleLocusEvent({locusUrl: 'test'});
+            assert.calledOnce(MeetingCollection.getByKey);
+            assert.calledWith(MeetingCollection.getByKey, 'locusUrl', 'test');
+            assert.calledOnce(parse);
+            assert.calledWith(parse, {locusInfo: {parse}}, {locusUrl: 'test'});
+          });
+        });
+        describe('there was not a meeting', () => {
+          let initialSetup;
+          beforeEach(() => {
+            initialSetup = sinon.stub().returns(true);
+            MeetingCollection.getByKey = sinon.stub().returns(undefined);
+            spark.meetings.create = sinon.stub().returns(Promise.resolve({locusInfo: {initialSetup}}));
+          });
+          it('should setup the meeting by difference event', async () => {
+            await spark.meetings.handleLocusEvent({locus: 'test', eventType: 'locus.difference', locusUrl: 'test'});
+            assert.calledOnce(MeetingCollection.getByKey);
+            assert.calledWith(MeetingCollection.getByKey, 'locusUrl', 'test');
+            assert.calledOnce(initialSetup);
+            assert.calledWith(initialSetup, 'test', {locusInfo: {initialSetup}});
+          });
+          it('should setup the meeting by difference event', async () => {
+            await spark.meetings.handleLocusEvent({locus: 'test', eventType: 'test', locusUrl: 'test'});
```

Comment:
this is purely an internal handling correct? an end user wouldn't invoke that method?

Reply:
correct